frontend/src/components/NewPassword/NewPassword.js

```diff
@@ -22,14 +27,34 @@ class NewPassword extends Component {
     this.setState({ [event.target.name]: event.target.value });
   };
 
+  handleSubmit = (event) => {
+    const { token } = this.props;
+    event.preventDefault();
+    const { password, confirmPassword } = this.state;
+    if (password !== confirmPassword) {
+      this.setState({ error: "Passwords don't match" });
+    } else {
+      resetPassword(token, password)
+        .then(this.handleRedirect)
+        .catch((error) => {
+          this.setState({ error: error.message });
+        });
+    }
+  };
+
+  handleRedirect = () => {
+    const { history } = this.props;
+    history.push('/login');
```

Comment:
How about we just cut out this function and move the `history.push` up into the `handleSubmit` function. We can pull the `history` prop at the same time as the `token` prop on line 31.